MEDMI_Data_Download/met_extraction_script.py

```diff
@@ -48,107 +163,237 @@ def _extract_data(self, dict, settings, extra_datasets=[]):
+
+    def _perform_extraction(self, dict):
+        datadata = Dataset(dict)
+        datadata.default()
+        return datadata
+
 
 
 class MetExtractorRain(MetExtractor):
-    def __init__(self, outfile_dir):
-        super(MetExtractorRain, self).__init__(outfile_dir)
-        self.source_reference = 'midas.rain_drnl_ob.prcp_amt 1'
-        self.head_string = 'Rain gauge daily data for date range: {} to {}\n'
-
-    def extract_data(self, date_range, latitude_range, longitude_range, outfile_suffix):
-        column_string = self.cols_base.format('rain')
-        filename = '{}/rain_{}.csv'.format(self.out_dir, outfile_suffix)
-        print('extracting {}'.format(self.head_string.format(date_range[0], date_range[1])))
-        dict = super(MetExtractorRain, self)._create_dict_base(date_range, latitude_range, longitude_range)
-        dict.update({'Source reference': self.source_reference})
-        settings = {'fname': filename,
-                    'headstring': self.head_string.format(date_range[0], date_range[1]),
-                    'columnstring': column_string}
-        self._extract_data(dict, settings)
+    SOURCE_REFERENCE = 'midas.rain_drnl_ob.prcp_amt 1'
+    MEASUREMENT_NAME = 'rain'
+
+    def __init__(self, out_dir=MetExtractor.DEFAULT_OUT_DIR, verbose=MetExtractor.DEFAULT_VERBOSE):
+        super(MetExtractorRain, self).__init__(out_dir, verbose)
+        self._head_string = 'Rain gauge daily data {} for date range: {} to {}\n'
+        self._cols_specific = [MetExtractorRain.MEASUREMENT_NAME]
+        self._filename = '{}/rain{}{}.csv'.format(self._out_dir, '{}', '{}')
+
 
 
 class MetExtractorTemp(MetExtractor):
-    def __init__(self, outfile_dir):
-        super(MetExtractorTemp, self).__init__(outfile_dir)
-        self.source_reference = 'midas.weather_hrly_ob.air_temperature'
-        self.head_string = 'Temperature, Relative Humidity, Station Pressure, and Wet Bulb Temperature hourly \
-                                    data for date range: {} to {}\n'
-
-    def extract_data(self, date_range, latitude_range, longitude_range, outfile_suffix):
-        column_string = self.cols_base.format('temperature,rh,pressure,wbtemp')
-        filename = '{}/temp_rh_press_wbulb_{}.csv'.format(new_dir, outfile_suffix)
-        extra_datasets = ['midas.weather_hrly_ob.rltv_hum', 'midas.weather_hrly_ob.stn_pres',
-                          'midas.weather_hrly_ob.dewpoint']
-        print('extracting {}'.format(self.head_string.format(date_range[0], date_range[1])))
-        dict = self._create_dict_base(date_range, latitude_range, longitude_range)
-        dict.update({'Source reference': self.source_reference})
-        settings = {'fname': filename,
-                    'headstring': self.head_string.format(date_range[0], date_range[1]),
-                    'columnstring': column_string}
-        self._extract_data(dict, settings, extra_datasets)
+    DEFAULT_PROCESS_SPATIAL_TEMPERATURE = 'sp_altadj_idw_mean'
```

Comment:
I don't think we should include the spatial processing (as this was for the pollen timeseries, which we aren't going to deal with anymore)?